tests/unit/socket_test.cc

```diff
@@ -546,3 +547,613 @@ SEASTAR_THREAD_TEST_CASE(inet_local_remote_address_sanity) {
     ss.wait_input_shutdown().get();
     BOOST_CHECK(ss.remote_address().is_unspecified());
 }
+
+// Comprehensive tests for proxy protocol v2 implementation
```

Comment:
When you add 16 tests for one feature (the proxy protocol), around 600 lines of code, I think it makes a lot of sense to put it in a separate source file.